### PR TITLE
add secret formats to readme

### PR DESCRIPTION
**Pull request by TalShorSap**
add secret formats to readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ The SAP BTP service operator is based on the [Kubernetes Operator pattern](https
 * [Credentials Rotation](#credentials-rotation)
 * [Multitenancy](#multitenancy)
 * [Troubleshooting and Support](#troubleshooting-and-support)
+* [Secret Formats](#secret-formats)
```
Comment (danielforsap) [inline README.md:30]: Maybe Formats of 'Secret' Objects

This way it seems like formats are secret.

File: README.md [hunk 2]
```diff
@@ -263,8 +264,8 @@ Review the supported Kubernetes API versions for the following SAP BTP Service O
 | serviceInstanceName`*`   | `string`   |  The Kubernetes name of the service instance to bind, should be in the namespace of the binding. |
 | externalName       | `string`   |  The name for the service binding in SAP BTP, defaults to the binding `metadata.name` if not specified. |
 | secretName       | `string`   |  The name of the secret where the credentials are stored, defaults to the binding `metadata.name` if not specified. |
-| secretKey | `string`  | The key inside the binding secret to store the credentials returned by the broker encoded as json to support complex data structures. |
-| secretRootKey | `string`  | The key inside the secret to store all binding data including credentials returned by the broker and additional info under single key.<br/>Convenient way to store whole binding data in single file when using `volumeMounts`. |
+| secretKey | `string`  | The key inside the binding secret to store the credentials returned by the broker encoded as json to support complex data structures. [Example](#secret-formats)|
```
Comment (danielforsap) [inline README.md:267]: Key is part of the Secret object that stores service binding data (credentials) returned from the broker. It is encoded as a JSON object to support complex data structures.

File: README.md [hunk 3]
```diff
@@ -263,8 +264,8 @@ Review the supported Kubernetes API versions for the following SAP BTP Service O
...
+| secretRootKey | `string`  | The key inside the secret to store all binding data including credentials returned by the broker and additional info under single key.<br/>Convenient way to store whole binding data in single file when using `volumeMounts`. [Example](#secret-formats)|
```
Comment (danielforsap) [inline README.md:268]: The root key is a part of the Secret object, which stores service binding data (credentials) received from the broker, as well as additional information. When the root key is used, all data is stored under a single key. This makes it a convenient way to store data in one file when using `volumeMounts`.

File: README.md [hunk 4]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
 
 [Back to top](#sap-business-technology-platform-sap-btp-service-operator-for-kubernetes)
 
+## Secret Formats
+
+### Default
+```bash
+#Credentials section
```
Comment (danielforsap) [inline README.md:525]: Credentials Section

File: README.md [hunk 5]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+
+#Additional data added by the operator
+instance_guid: bd9bda65-1405-4ea0-a716-5afeb13eca5b
+instance_name: sample-instance    
+plan: lite                        
+type: feature-flags  // The service offering name
+```
+
+### SecretKey
+Done by adding to the spec -> secretKey: YourSecretValue
```
Comment (danielforsap) [inline README.md:538]: To use SecretKey, add 'secretKey: your secret value' to the spec.  (add <> to your secret value, won't show in the comments.

File: README.md [hunk 6]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+```bash
+#Credentials section
```
Comment (danielforsap) [inline README.md:540]: Credentials Section

File: README.md [hunk 7]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+<YourSecretValue>:
```
Comment (danielforsap) [inline README.md:541]: Maybe  <your secret value> ?

Comment (danielforsap) [reply README.md:541]: your secret value but with < >  
Won't show in the comments.

File: README.md [hunk 8]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+
+#Additional data added by the operator
+instance_guid: bd9bda65-1405-4ea0-a716-5afeb13eca5b
+instance_name: sample-instance  
+plan: lite
+type: feature-flags // The service offering name
+```
+
+### SecretRootKey
+Done by adding to the spec -> secretRootKey: YourSecretValue
```
Comment (danielforsap) [inline README.md:556]: To use SecretRootKey, add 'secretRootKey: your secret value' to the spec.  (add <> to your secret value, won't show in the comments.

File: README.md [hunk 9]
```diff
@@ -517,6 +518,60 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+```bash
+<YourSecretValue>:
+  {
+    #Credentials section
```
Comment (danielforsap) [inline README.md:560]: Credentials Section

**Review by danielforsap — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (TalShorSap) [reply README.md:30]: done

File: README.md (hunk 2 above)
Comment (TalShorSap) [reply README.md:267]: done

File: README.md (hunk 3 above)
Comment (TalShorSap) [reply README.md:268]: done

File: README.md (hunk 4 above)
Comment (TalShorSap) [reply README.md:525]: I want to be clear that this is a comment

File: README.md (hunk 6 above)
Comment (TalShorSap) [reply README.md:540]: same

File: README.md (hunk 7 above)
Comment (TalShorSap) [reply README.md:541]: done

**Review by danielforsap — APPROVED** [review]
After UA review.

File: README.md [hunk 10]
```diff
@@ -258,20 +259,20 @@ spec:
...
+| externalName       | `string`   | The name for the service binding in SAP BTP, defaults to the binding `metadata.name` if not specified.                                                                                                                                                                                                                                   |
+| secretName       | `string`   | The name of the secret where the credentials are stored, defaults to the binding `metadata.name` if not specified.                                                                                                                                                                                                                       |
+| secretKey | `string`  | Property of the Secret object that stores service binding data (credentials) returned from the broker. It is encoded as a JSON object to support complex data structures. [Example](#formats-of-secret-objects)                                                                                                                                     |
```
Comment (avilupu) [inline README.md:267]: In rootKey you use 'property' while in secretRootKey we use 'key'. key is also baked in the attributes' name. And you also use it below in the format title ("Key- Value Pairs") and the description.

"...defines a name of a key in the secret object in which the service binding data (credentials) returned from the broker will be stored..."

File: README.md [hunk 11]
```diff
@@ -258,20 +259,20 @@ spec:
...
+| secretRootKey | `string`  | The root key is a part of the Secret object, which stores service binding data (credentials) received from the broker, as well as additional information. When the root key is used, all data is stored under a single key. This makes it a convenient way to store data in one file when using volumeMounts. [Example](#formats-of-secret-objects) |
```
Comment (avilupu) [inline README.md:268]: This is a good description. we can use similar phrasing for the secretKey (see comment above).

File: README.md [hunk 12]
```diff
@@ -517,6 +518,66 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+
+#Service instance info
+instance_guid: <instance_guid> // The service instance ID
+instance_name: my-service-btp-name // Taken from the service instance external_name field if set. Otherwise from metadata.name
+plan: sample-plan // The service plan name                
+type: sample-service  // The service offering name
+```
+
+### Credentials as JSON Object
+To show credentials returned from the broker as a JSON object, add 'secretKey: your-secret-value' to the binding spec.
```
Comment (avilupu) [inline README.md:539]: it shoud be clearer that 'your-secret-value' is a sample value.
"add 'secretKey: your-secret-value' to the binding spec" seems like straight forward instruction to follow.

"...specify your desired key name in add 'secretKey' attribute in the binding spec"

File: README.md [hunk 13]
```diff
@@ -517,6 +518,66 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+type: sample-service  // The service offering name
+```
+
+### Credentials as JSON Object
+To show credentials returned from the broker as a JSON object, add 'secretKey: your-secret-value' to the binding spec. 
+
+your-secret-vaue will be a key for a value in JSON format that contains credentials while service instance info remains presented in key-value pairs.
+```bash
+#Credentials
+your-secret-value:
```
Comment (avilupu) [inline README.md:544]: it is a value/name of a key, not a secret 
use 'my-key' for example...

File: README.md [hunk 14]
```diff
@@ -517,6 +518,66 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+}
+
+#Service Instance info
+instance_guid: <instance_guid> // The service instance ID
+instance_name: my-service-btp-name // Taken from the service instance external_name field if set. Otherwise from metadata.name 
+plan: sample-plan // The service plan name
+type: sample-service // The service offering name
+```
+
+## Credentials and Binding Info as One JSON Object
```
Comment (avilupu) [inline README.md:558]: why binding info? in the sanples and description we use instance info

File: README.md [hunk 15]
```diff
@@ -517,6 +518,66 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+To show both credentials returned from the broker and service instance info as a JSON object, add 'secretRootKey: your-secret-value' to the binding spec.
```
Comment (avilupu) [inline README.md:559]: same as above

File: README.md [hunk 16]
```diff
@@ -517,6 +518,66 @@ The SAP BTP service operator project maintainers will respond to the best of the
...
+
+your-secret-value becomes a key that contains credentials and service instance info in a JSON format.
+
+```bash
+your-secret-value:
```
Comment (avilupu) [inline README.md:564]: same a above

File: README.md (hunk 10 above)
Comment (danielforsap) [reply README.md]: @TalShorSap 
Part of the Secret object that stores service binding data (credentials) received from the broker. When secretKey is used, credentials are stored in the JSON format.

File: README.md (hunk 14 above)
Comment (danielforsap) [reply README.md:558]: @avilupu  apparently this change was overridden. 
@TalShorSap

File: README.md (hunk 12 above)
Comment (danielforsap) [reply README.md:539]: To show credentials returned from the broker as a JSON object, specify your desired key name for 'secretKey' attribute in the binding spec. 'secretKey: your-desired-key-name'

File: README.md (hunk 14 above)
Comment (TalShorSap) [reply README.md:558]: changed

File: README.md (hunk 12 above)
Comment (TalShorSap) [reply README.md:539]: added

File: README.md (hunk 11 above)
Comment (TalShorSap) [reply README.md]: done

File: README.md (hunk 10 above)
Comment (TalShorSap) [reply README.md]: done

File: README.md (hunk 13 above)
Comment (TalShorSap) [reply README.md:544]: ok changing

File: README.md (hunk 15 above)
Comment (TalShorSap) [reply README.md:559]: changed

File: README.md (hunk 12 above)
Comment (TalShorSap) [reply README.md]: @danielforsap need to add reason and take care for Avi's comment

File: README.md (hunk 10 above)
Comment (TalShorSap) [reply README.md]: @danielforsap 
The root key... -> The secretRootKey is the key in the secret object in which... (value is json)

**Pull request merged**